tests/src/Integration/Engine/ComponentActionTest.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Tests\rules\Integration\Engine\ComponentActionTest.
+ */
+
+namespace Drupal\Tests\rules\Integration\Engine;
+
+use Drupal\Core\Config\Entity\ConfigEntityStorageInterface;
+use Drupal\rules\Entity\RulesComponentConfig;
+use Drupal\Tests\rules\Integration\RulesEntityIntegrationTestBase;
+
+/**
+ * Tests for exposing Rules components as action plugins.
+ *
+ * @group rules
+ */
+class ComponentActionTest extends RulesEntityIntegrationTestBase {
+
+  /**
+   * Tests that a rule can be used as action.
+   */
+  public function testActionAvailable() {
+    $rule = $this->rulesExpressionManager->createRule();
+
+    $rules_config = new RulesComponentConfig([
+      'id' => 'test_rule',
+      'label' => 'Test rule',
+    ], 'rules_component');
+    $rules_config->setExpression($rule);
+
+    $storage = $this->prophesize(ConfigEntityStorageInterface::class);
```

Comment:
We can add an additional kernel test for config dependencies, but for testing derivative and context definition functionality this integrity test should be fine and is much faster to execute.